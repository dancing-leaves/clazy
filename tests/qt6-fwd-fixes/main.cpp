#include <QtCore/qglobal.h>

class myClass
{
  class QStringList; // this is ignored
};

template <typename Key, typename T> class QCache  ;     class SomeClass;
template <typename Key, typename T> class QHash  ;
template <typename Key, typename T> class QMap /*blabla*/ ;
template <typename Key, typename T> class QMultiHash;
template <typename Key, typename T> class QMultiMap;
class QPair;
template <typename T> class QQueue;
template <typename T> class QSet;
template <typename T> class QStack;
template <typename T, qsizetype Prealloc = 256> class QVarLengthArray;
template <typename T> class QList;
template<typename T> class QVector;
class QStringList;
class QByteArrayList;
class QMetaType;
class QVariant;
class QVariantList;
class QVariantMap;
class QVariantHash;
class QVariantPair;
