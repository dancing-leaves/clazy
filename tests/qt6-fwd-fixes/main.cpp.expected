qt6-fwd-fixes/main.cpp:8:1: warning: Using forward declaration of QCache. Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:9:1: warning: Using forward declaration of QHash. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:10:1: warning: Using forward declaration of QMap. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:11:1: warning: Using forward declaration of QMultiHash. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:12:1: warning: Using forward declaration of QMultiMap. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:13:1: warning: Using forward declaration of QPair. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:14:1: warning: Using forward declaration of QQueue. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:15:1: warning: Using forward declaration of QSet. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:16:1: warning: Using forward declaration of QStack. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:17:1: warning: Using forward declaration of QVarLengthArray. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:18:1: warning: Using forward declaration of QList. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:19:1: warning: Using forward declaration of QVector. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:20:1: warning: Using forward declaration of QStringList. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:21:1: warning: Using forward declaration of QByteArrayList. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:22:1: warning: Using forward declaration of QMetaType. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:23:1: warning: Using forward declaration of QVariant. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:24:1: warning: Using forward declaration of QVariantList. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:25:1: warning: Using forward declaration of QVariantMap. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:26:1: warning: Using forward declaration of QVariantHash. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
qt6-fwd-fixes/main.cpp:27:1: warning: Using forward declaration of QVariantPair. (already) Including <QtCore/qcontainerfwd.h> instead. [-Wclazy-qt6-fwd-fixes]
